lib/addons/jupyterhub/index.ts

```diff
@@ -67,64 +105,206 @@ export class JupyterHubAddOn extends HelmAddOn {
         this.options = this.props as JupyterHubAddOnProps;
     }
 
-    @dependable('EbsCsiDriverAddOn')
+    @dependable('EbsCsiDriverAddOn' || 'EfsCsiDriverAddOn')
```

Comment:
You cannot define dependency with a logic OR. What do you think the result is when a boolean operator is applied to strings? Tbh, I would expect a compile error, but may be ts just returns one of the expressions.

You can pass both of the addons separated by `,`  or one of the addons. 

We can extend the dependable decorator to accept a comparison custom function like `(a)=> return a == 'A' or a.equals('B')`

Reply:
Hmm - I just tested this and it only checks the first one (EBS). I will make a change on the decorator to accept a comparison.

Reply:
You can also create two methods in the addons: addEbsStorage and addEbsStorage each with its own decorator. That would be the easiest of each storage method translated in a construct.

You can also add a manual validation using `clusterInfo.getScheduledAddOn` to test.

Decorator while convenient and handy, is the least direct approach: you will still be able to define ebsConfig add dependency on EFS driver and fail.